lib/semantics/semantics.cc

```diff
@@ -168,6 +298,24 @@ Scope &SemanticsContext::FindScope(parser::CharBlock source) {
   }
 }
 
+void SemanticsContext::PushConstruct(const ConstructNode &construct) {
+  constructStack_.emplace_back(construct);
+}
+
+void SemanticsContext::PopConstruct() {
+  CHECK(!constructStack_.empty());
```

Comment:
This `CHECK` may not add any value; the `pop_back()` will crash if the stack is empty.

Reply:
```CHECK``` has the advantage that the message coming from the compiler contains that name of the file and source line of the failing check.  When ```pop_back()``` fails, the compiler just produces a segmentation fault with no indication of the location of the failure.